compiler/src/test-integration/java/io/neow3j/compiler/StringConcatenationTest.java

```diff
@@ -64,6 +98,31 @@ public static String concatInStaticVariable() {
             return staticString;
         }
 
+        public static String concatStringWithMethodReturn() {
```

Comment:
does it worth adding tests with all Java built-in raw types?

Like: `int` (we already have test for this), `byte`, `short`, `long`, `float`, `double`, `boolean`, and `char`? 

Also, maybe, array of these types?

wdyt?

Reply:
@gsmachado 
- Trying to concatenate arrays (except `byte[]`) results in an exception on the NeoVM (`Specified cast is not valid`). Therefore, no array types can be used in string concatenations in neow3j. I have added a UT for this in the class `StringLiteralHelperTest.java`.

- `byte`, `byte[]`, `long`, `short` all have the same behaviour as `int`. There are UTs in place for `int` concatenation with strings that are expected to throw an exception. Hence, there's no need to have more tests for these types atm.

- `char` works the same way as `String`. I've added integration tests for the primitive type as well as for the wrapper class (`java/lang/Character`).

- Last but not least: Neow3j does not support float and double in the first place. So, no need for further tests there. The compilation would fail as soon as it notices such types present anyways.

Reply:
great! All fine, then.